fjord/feedback/tests/test_views.py

```diff
@@ -115,68 +117,60 @@ def test_valid_sad(self):
         eq_(u'14.0.1', feedback.version)
 
     @with_waffle('thankyou', True)
-    def test_use_thank_you_sad(self):
-        """Thanks_sad template is used when appropriate"""
+    def test_thankyou_flag_active(self):
+        """Verify response and suggestions when thankyou flag is active"""
         url = reverse('feedback', args=(u'firefox',), locale='en-US')
         r = self.client.post(url, {
             'happy': 0,
-            'description': u"I want to know why Firefox doesn't make me sandwiches!",
+            'description': u'Why Firefox not make me sandwiches!',
         }, follow=True)
 
         feedback = models.Response.objects.latest(field_name='id')
-        eq_(u"I want to know why Firefox doesn't make me sandwiches!", feedback.description)
-        eq_(u'', feedback.url)
-        eq_(False, feedback.happy)
-        eq_(u'en-US', feedback.locale)
         eq_(r.status_code, 200)
-        self.assertTemplateUsed(r, 'feedback/thanks_sad.html')
+        eq_(r.context['response'].id, feedback.id)
+        eq_(r.context['suggestions'], [])
 
     @with_waffle('thankyou', True)
-    def test_use_default_thank_you_words(self):
-        """Default thank you page is used when descirption is under 7 words"""
-        url = reverse('feedback', args=(u'firefox',), locale='en-US')
-        r = self.client.post(url, {
-            'happy': 0,
-            'description': u"Why doesn't it load?",
-        }, follow=True)
+    def test_thankyou_flag_and_response_id_in_qs_unauthenticated(self):
+        """Verify response_id in querystring is ignored if user is not
+        authenticated
+        """
+        feedback = ResponseFactory()
+        url = reverse('thanks') + '?response_id={0}'.format(feedback.id)
+        r = self.client.get(url)
 
-        feedback = models.Response.objects.latest(field_name='id')
-        eq_(u'', feedback.url)
         eq_(r.status_code, 200)
-        self.assertTemplateUsed(r, 'feedback/thanks.html')
+        eq_(r.context['response'], None)
+        eq_(r.context['suggestions'], None)
 
     @with_waffle('thankyou', True)
-    def test_use_default_thank_you_happy(self):
-        """Default thank you page is used when the feedback is happy."""
+    def test_thankyou_flag_and_response_id_in_qs_authenticated(self):
+        """Verify response_id in querystring overrides session id"""
+        # Create analyzer and log analyzer in
+        jane = ProfileFactory(user__email='jane@example.com').user
+        jane.groups.add(Group.objects.get(name='analyzers'))
```

Comment:
This code is fine, I don't think you need to change it. I'm curious though, could FactoryBoy make a `AnalyzerProfileFactory` that did this?